lib/AST/ASTVerifier.cpp

```diff
@@ -988,6 +988,13 @@ class Verifier : public ASTWalker {
       }
       
       if (S->hasResult()) {
+        if (isa<ConstructorDecl>(func)) {
+          Out << "Expected ReturnStmt not to have a result. A constructor should not return a result. Returned expression: ";
+          S->getResult()->dump(Out);
+          Out << "\n";
+          abort();
```

Comment:
Just to double-check, `return nil` in failable constructors does not trigger this new error, right? It should be covered by existing tests I think, but worth checking.

Reply:
A failable constructor will return an `Optional<Class>` where the element is `.none`. So, this error is not triggered (also, for good measure, I ran a test program and it didn't trigger this error).